api/store/create.js

```diff
@@ -0,0 +1,93 @@
+module.exports = createStore
+
+var _ = require('lodash')
+var request = require('request').defaults({json: true})
+
+var addRolePrivilege = require('../utils/add-role-privilege')
+var couchDbDesignDoc = require('../utils/couchdb-design-doc')
+var errors = require('../utils/errors')
+var getRoleOption = require('../utils/get-role-option')
+var hoodieAccessToCouchDbSecurity = require('../utils/hoodie-access-to-couchdb-security')
+var storeExists = require('./exists')
+var toDbId = require('../utils/to-db-id')
+
+function createStore (state, name, options) {
+  return storeExists(state, name)
+
+  .then(function (doesExist) {
+    if (doesExist) {
+      throw errors.CONFLICT
+    }
+
+    var doc = {
+      _id: toDbId(name),
+      access: {
+        read: {
+          role: []
+        },
+        write: {
+          role: []
+        }
+      }
+    }
+
+    if (options) {
+      _.concat(options.access).forEach(function (privilege) {
+        addRolePrivilege(doc.access, getRoleOption(options), privilege)
+      })
+    }
+
+    return state.metaDb.put(doc)
+
+    .then(function () {
+      var db = new state.PouchDB(name)
```

Comment:
I think that considering we use `encodeURIComponent` elsewhere in this code, we should possibly use it here too - at least in the short term. Just avoids any work-arounds. I also tried running `encodeURIComponent` before being passed to `Store.create`, and got the error `Name: 'entity%2F4797965e8392762f3331ab0913037f61%2Fclients'. Only lowercase characters (a-z), digits (0-9), and any of the characters _, $, (, ), +, -, and / are allowed. Must begin with a letter.`